inference-engine/src/gna_plugin/memory/gna_memory_state.hpp

```diff
@@ -11,16 +11,27 @@
 
 namespace  GNAPluginNS {
 namespace memory {
-class GNAMemoryState : public InferenceEngine::MemoryStateInternal {
-    std::shared_ptr<GNAPlugin> plg;
+class GNAMemoryState : public InferenceEngine::IMemoryStateInternal {
  public:
-    using Ptr = InferenceEngine::MemoryStateInternal::Ptr;
+    GNAMemoryState(std::string name, GNAMemoryLayer *state)
+        : name(name), state(state) { }
```

Comment:
most of functions rely on state != nullptr without explicit check - either add assert here or in every function - why not use Weak or Shared ptr here?